src/routes/CustomAsset/sagas/customAsset.js

```diff
@@ -1,151 +1,224 @@
+  const { result } = yield race({
+    result: requestAssetFieldsTask(contractAddress),
+    // cancel on close Add/Edit asset screen
+    close: take(CLOSE),
+    // cancel, when we are trying to request another contract
+    restart: take(START_ASSET_LOADING),
+  })
 
-    yield put(editSuccess(updatedAssets))
-  } catch (err) {
-    yield put(editError(err))
+  if (result) {
+    yield put(setAssetIsValid(true))
   }
+
+  return result
 }
 
-function* removeCustomAsset(): Saga<void> {
+/**
+ * Fires, when user changes fields on CustomAssetForm
+ * Form validation here
+ * Emit setField, setFieldError, clearFieldError, requestAssetFields events
+ *
+ * @param action from setField method
+ * @returns {undefined}
+ */
+function* onFieldChange(action: ExtractReturn<typeof setField>): Saga<void> {
+  const { fieldName, value } = action.payload
+
   try {
-    const digitalAssets: DigitalAssets = yield select(selectDigitalAssetsItems)
-    const { address }: CustomAssetData = yield select(selectCustomAsset)
+    switch (fieldName) {
+      /**
+       * Address field validation
+       * This input is avalable only in add asset page, in edit asset is is disabled
+       */
+      case 'address': {
+        const contractAddress = value.trim()
+
+        // 0. clear error when address is empty
+        if (contractAddress === '') {
+          yield put(clearFieldError('address'))
+          break
+        }
 
-    const updatedAssets: DigitalAssets = digitalAssets
-      .filter((asset: DigitalAsset): boolean => (asset.address !== address))
+        // 1. simple checks
+        addressFieldSimpleCheck(contractAddress)
 
-    yield put(removeSuccess(updatedAssets))
-  } catch (err) {
-    yield put(removeError(err))
-  }
-}
+        // 2. check if this asset already exists
+        const foundAsset = yield* getDigitalAsset(contractAddress)
+        if (foundAsset) {
+          throw new InvalidFieldError('address', i18n('general.error.address.exists'))
+        }
 
-function getCustomAssetData(customAssetData: CustomAssetData): DigitalAsset {
-  const { name, symbol, address, decimals }: CustomAssetData = customAssetData
+        // 4. Request name, symbol, decimals fields from contract
+        const assetFields = yield* requestAssetFields(contractAddress)
 
-  return {
-    name,
-    symbol,
-    address,
-    decimals: parseInt(decimals, 10) || 0,
-    isActive: true,
-    isCustom: true,
-  }
-}
+        // update form fields
+        if (assetFields) {
+          yield put(setField('name', assetFields.name))
+          yield put(setField('symbol', assetFields.symbol))
+          yield put(setField('decimals', assetFields.decimals))
+        }
 
-export function* watchCustomAssetClose(): Saga<void> {
-  yield takeEvery(CLOSE, closeAddCustomAsset)
-}
+        break
+      }
 
-export function* watchCustomAssetAdd(): Saga<void> {
-  yield takeEvery(ADD, addCustomAsset)
-}
+      // reserved for add, edit asset task
```

Comment:
always better to avoid to commit such stuff, I understand the purpose. But it can be handled by other developer and in different way.